File: src/app/classes/all-results-response.ts
import { Result } from "../classes/result"

export class AllResultsResponse {
    count: number;
    next?: string | null;
    previous?: string | null;
    results: Result[];

    constructor(count: number, results: Result[]) {
        this.count = count;
        this.results = JSON.parse(JSON.stringify(results));
    }
}

File: back/src/main/java/com/pokemon/api/model/responses/AllResultsResponse.java
package com.pokemon.api.model.responses;

import com.pokemon.api.model.ResultModel;
import lombok.Data;

import java.util.List;

@Data
public class AllResultsResponse {
    private int count;
    private String next;
    private String previous;
    private List<ResultModel> results;

    public AllResultsResponse(int count, List<ResultModel> results) {
        this.count = count;
        this.results = results;

        this.next = "";
        this.previous = "";
    }
}